### PR TITLE
Proposals

### PR DESCRIPTION
**Pull request by th3o6a1d**
Big contract upgrades
```
newProject(metadataURI, owners, finder, threshold) -- creates new project proposal and pushes to a mapping.

getProjectIndex() -- returns last created project id

getProject(id) -- returns status of project (votes, whether it's been deployed, metadata URI)

getProjects(startindex,endindex) -- returns array of projects

vote(id, boolean) -- vote yes or no (true/false)

getVoters(id)-- see who has voted (but not how, at this point)

getVote(id, voter) -- how a person voted for a specific project

deployProject(id) -- calls createProject to deploy a project that has votes and is passing. 
```
See `tests.js`

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,4 +1,4 @@
...
..."address" } ], "name": "release", "outputs": [], "stateMutability": "nonpayable", "type": "function" }, { "inputs": [ { "internalType": "address", "name": "account", "type": "address" } ], "name": "released", "outputs": [ { "internalType": "uint256", "name": "", "type": "uint256" } ], "stateMutability": "view", "type": "function" }, { "inputs": [ { "internalType": "address", "name": "account", "type": "address" } ], "name": "shares", "outputs": [ { "internalType": "uint256", "name": "", "type": "uint256" } ], "stateMutability": "view", "type": "function" }, { "inputs": [], "name": "totalReleased", "outputs": [ { "internalType": "uint256", "name": "", "type": "uint256" } ], "stateMutability": "view", "type": "function" }, { "inputs": [], "name": "totalShares", "outputs": [ { "internalType": "uint256", "name": "", "type": "uint256" } ], "stateMutability": "view", "type": "function" }, { "stateMutability": "payable", "type": "receive" }, { "inputs": [ { "internalType": "address[]", "name": "_payees", "type": "address[]" }, { "internalType": "uint256[]", "name": "_shares", "type": "uint256[]" } ], "name": "initialize", "outputs": [], "stateMutability": "payable", "type": "function" } ]
```
Comment (carlosfebres) [inline .env]: @th3o6a1d we don't need to have these ABI in ENVs as they don't depend on the environment. we can move them to a typescript file.
Also, as @cbrzn recommended, let's convert these ABI to [Human-Readable](https://docs.ethers.io/v5/api/utils/abi/formats/#abi-formats--human-readable-abi). you can follow [this guide](https://docs.ethers.io/v5/api/utils/abi/formats/#abi-formats--converting-between-formats) to make the transformation.

**Review by carlosfebres — COMMENTED** [review]
Great work man! I'll work to integrate these contracts 💪🏻

File: .env (hunk 1 above)
Comment (benefacto) [reply .env]: FYI, one alternative way of consuming ABIs is using [TypeChain](https://github.com/ethereum-ts/Typechain).

File: .gitignore [hunk 2]
```diff
@@ -30,3 +30,5 @@ yarn-debug.log*
 yarn-error.log*
 
 /src/contracts/models
+
+package-lock.json
```
Comment (benefacto) [inline .gitignore]: The `package-lock,json` should absolutely not be added to the .gitignore. Please see [this](https://stackoverflow.com/a/48524475) for more info.

File: contracts/dOrgProjectFactory.sol [hunk 3]
```diff
@@ -22,31 +22,134 @@ interface GnosisSafe {
...
+            require(Voters[i][v] != msg.sender, 'Cannot vote twice.');
+        }
+        require(
+            block.timestamp < Projects[i].createdAt + 604800,
+            "Voting period has closed."
+        );
+        if (approval == true) {
+            Projects[i].forVotes += 1;
+        }
+        if (approval == false) {
```
Comment (benefacto) [inline contracts/dOrgProjectFactory.sol]: This is a bit nitpicky but this could be just an `else` statement.

**Review by benefacto — COMMENTED** [review]
I left some minor comments but this looks good to me! (FYI, I'm not a senior smart contract dev)

File: .env (hunk 1 above)
Comment (th3o6a1d) [reply .env]: Good call. Removed these ABIs from the `.env` file. Now it's just the rinkeby contract location.

File: .gitignore (hunk 2 above)
Comment (th3o6a1d) [reply .gitignore]: Interesting. Makes sense. will remove from `.gitignore`. This is why we code review :)

File: contracts/dOrgProjectFactory.sol (hunk 3 above)
Comment (th3o6a1d) [reply contracts/dOrgProjectFactory.sol]: True. Edited this.

**Pull request merged**